pkg/solana/chain.go

```diff
@@ -164,8 +178,13 @@ func (v *verifiedCachedClient) verifyChainID(ctx context.Context) (bool, error)
 		return v.chainIDVerified, fmt.Errorf("failed to fetch ChainID in verifiedCachedClient: %w", err)
 	}
 
-	// if this is localnet, allow any chain ID as long as it's not spoofing an official network
-	if v.expectedChainID != "localnet" {
+	_, err = chainsel.GetChainDetailsByChainIDAndFamily(v.expectedChainID, chainsel.FamilySolana)
```

Comment:
Can we instead add this basic check at chain startup itself?
If a NOP has bad config, we should just fail to start the chain, rather than fail later at this stage.

I mean here:
https://github.com/smartcontractkit/chainlink-solana/blob/e8b0555aed77a2c17f073062c42c9090a617a81d/pkg/solana/chain.go#L261-L263

Reply:
Yeah that's even better!

Reply:
If this provided chainId isn't in chain-selectors, then return error directly from newChain()

Reply:
This might not work because for unit test that uses localnet